src/canvasdrawer.js

```diff
@@ -0,0 +1,1118 @@
+                        degrees: this.viewport.getRotation(true),
+                        useSketch: false
+                    });
+                }
+                if (tiledImage.getRotation(true) % 360 !== 0) {
+                    this._offsetForRotation({
+                        degrees: tiledImage.getRotation(true),
+                        point: this.viewport.pixelFromPointNoRotate(
+                            tiledImage._getRotationPoint(true), true),
+                        useSketch: false
+                    });
+                }
+            }
+            this.blendSketch({
+                opacity: tiledImage.opacity,
+                scale: sketchScale,
+                translate: sketchTranslate,
+                compositeOperation: tiledImage.compositeOperation,
+                bounds: bounds
+            });
+            if (sketchScale) {
+                if (tiledImage.getRotation(true) % 360 !== 0) {
+                    this._restoreRotationChanges(false);
+                }
+                if (this.viewport.getRotation(true) % 360 !== 0) {
+                    this._restoreRotationChanges(false);
+                }
+            }
+        }
+
+        if (!sketchScale) {
+            if (this.viewport.getRotation(true) % 360 === 0 &&
+                tiledImage.getRotation(true) % 360 === 0) {
+                if(this.viewer.viewport.getFlip()) {
+                    this._flip();
+                }
+            }
+        }
+
+        this._drawDebugInfo( tiledImage, lastDrawn );
+
+        // Fire tiled-image-drawn event.
+        if( this.viewer ){
+            /**
+                * Raised when a tiled image is drawn to the canvas. Only valid
+                * for webgl and canvas drawers.
+                *
+                * @event tiled-image-drawn
+                * @memberof OpenSeadragon.Viewer
+                * @type {object}
+                * @property {OpenSeadragon.Viewer} eventSource - A reference to the Viewer which raised the event.
+                * @property {OpenSeadragon.TiledImage} tiledImage - Which TiledImage is being drawn.
+                * @property {Array} tiles - An array of Tile objects that were drawn.
+                * @property {?Object} userData - Arbitrary subscriber-defined object.
+                */
+            this.viewer.raiseEvent( 'tiled-image-drawn', {
+                tiledImage: tiledImage,
+                tiles: lastDrawn,
+            });
+        }
+    }
+
+    /**
+         * This function converts the given point from to the drawer coordinate by
+         * multiplying it with the pixel density.
+         * This function does not take rotation into account, thus assuming provided
+         * point is at 0 degree.
+         * @private
+         * @param {OpenSeadragon.Point} point - the pixel point to convert
+         * @returns {OpenSeadragon.Point} Point in drawer coordinate system.
+         */
+    _viewportCoordToDrawerCoord(point) {
```

Comment:
Wait a second... This looks like it's the same as `DrawerBase.viewportCoordToDrawerCoord`, except it has an underscore. Seems like you should get rid of this one and use the one in DrawerBase.

Reply:
Looks like I started the process of consolidating this into the parent class (since it is used in multiple child classes) and got derailed. Removed these private versions.